### ISSUE TITLE
IntegrationValidator.validate throws Caused by: java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by mochadwi** [BranchMetrics/android-branch-deep-linking-attribution#944]
Branch 5.0.14

`SplashActivity.kt` was calling the `IntegrationValidator.validate(this)` inside `onStart()` below crash occurred instead:

<details>
<summary>
<b>Caused by: java.util.ConcurrentModificationException</b>
</summary>

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.staging, PID: 4739
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.sampingan.agentapp.staging/com.sampingan.agentapp.activities.SplashActivity}: java.util.ConcurrentModificationException
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2757)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2818)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:163)
        at android.app.ActivityThread.main(ActivityThread.java:6393)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:933)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:823)
     Caused by: java.util.ConcurrentModificationException
        at java.util.LinkedHashMap$LinkedHashIterator.nextEntry(LinkedHashMap.java:429)
        at java.util.LinkedHashMap$KeyIterator.next(LinkedHashMap.java:440)
        at io.branch.referral.validators.BranchIntegrationModel.updateDeepLinkSchemes(BranchIntegrationModel.java:60)
        at io.branch.referral.validators.BranchIntegrationModel.<init>(BranchIntegrationModel.java:46)
        at io.branch.referral.validators.IntegrationValidator.<init>(IntegrationValidator.java:27)
        at io.branch.referral.validators.IntegrationValidator.validate(IntegrationValidator.java:32)
        at com.SplashActivity.onStart(SplashActivity.kt:149)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1249)
        at android.app.Activity.performStart(Activity.java:6884)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2720)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2818) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1557) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:163) 
        at android.app.ActivityThread.main(ActivityThread.java:6393) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:933) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:823) 
```
</details>

**Comment by jf-branch** [comment]
Hi @mochadwi, 

Thanks for the insight, we are able to repro on our side and will include a fix to this with our next update.

**Comment by agerstner-branch** [comment]
Addressed with 5.0.15. Thanks for reporting, please open a new issue if problem persists.
